pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -87,17 +87,25 @@ var (
 		  oc adm must-gather --image=my/image:tag --source-dir=/pod/directory -- myspecial-command.sh
 	`)
 
+	// According to https://github.com/openshift/enhancements/blob/master/enhancements/oc/must-gather.md#proposal
+	// we can assume /usr/bin/gather is always run in the end, even when a custom command is specified.
+	// That's why we always kill /usr/bin/gather, even though a custom command can be specified.
 	volumeUsageCheckerScript = `
-echo "volume percentage checker started....."
+echo "[disk usage checker] Started"
+target_dir="%s"
+usage_percentage_limit="%d"
 while true; do
-disk_usage=$(du -s "%s" | awk '{print $1}')
-disk_space=$(df -P "%s" | awk 'NR==2 {print $2}')
+disk_usage=$(du -s "$target_dir" | awk '{print $1}')
+disk_space=$(df -P "$target_dir" | awk 'NR==2 {print $2}')
 usage_percentage=$(( (disk_usage * 100) / disk_space ))
-echo "volume usage percentage $usage_percentage"
-if [ "$usage_percentage" -gt "%d" ]; then
-	echo "Disk usage exceeds the volume percentage of %d for mounted directory. Exiting..."
-	# kill gathering process in gather container to prevent disk to use more.
-	pkill --signal SIGKILL -f %s
+echo "[disk usage checker] Volume usage percentage: current = ${usage_percentage} ; allowed = ${usage_percentage_limit}"
+if [ "$usage_percentage" -gt "$usage_percentage_limit" ]; then
+	echo "[disk usage checker] Disk usage exceeds the volume percentage of ${usage_percentage_limit} for mounted directory, terminating..."
```

Comment:
SIGTERM silently terminates the process (logs are ambiguous. Because there is only written as `terminated`). I think, we need to send direct SIGKILL signal to kill the process without needing any graceful shutdown. Node disk size is under pressure. So that to protect other critical components, sending SIGKILL would make sense to me.

I'm suggesting this;

```sh
echo "[disk usage checker] Disk usage exceeds the volume percentage of ${usage_percentage_limit} for mounted directory, terminating..."
pkill --signal SIGKILL -f /usr/bin/gather
exit 1
```

I tested based on this and it worked nice in different scenarios.

Reply:
Amended.